sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/FilterPushDownSupport.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources.v2.reader;
+
+import org.apache.spark.sql.sources.Filter;
+
+public interface FilterPushDownSupport {
+  // TODO: shall we introduce the concept of "unhandledFilters"? Personally I feel it's not very
+  // useful and we can always execute the filter again at Spark side.
```

Comment:
According to the public `Filter` interface, we can only push down very simple filters, so this should not have performance issues. But we may need to add `unhandledFilters` to `CatalystFilterPushDownSupport`, which support arbitrary expressions and may have very expensive predicates.